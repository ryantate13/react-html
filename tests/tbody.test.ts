import {tbody} from '../src';

describe('tbody', () => {
    it('renders with children', () => {
        expect(tbody({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(tbody({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(tbody({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(tbody()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(tbody({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(tbody({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(tbody({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(tbody({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(tbody({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(tbody({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(tbody({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(tbody({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(tbody({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(tbody({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(tbody({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(tbody({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(tbody({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(tbody({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(tbody({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(tbody({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(tbody({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(tbody({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(tbody({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(tbody({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(tbody({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(tbody({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(tbody({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(tbody({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(tbody({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(tbody({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(tbody({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(tbody({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(tbody({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(tbody({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(tbody({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(tbody({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(tbody({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(tbody({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(tbody({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(tbody({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(tbody({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(tbody({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(tbody({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(tbody({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(tbody({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(tbody({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(tbody({translate: "no"})).toMatchSnapshot();
    });
});