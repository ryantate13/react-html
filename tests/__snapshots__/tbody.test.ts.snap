// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tbody renders with accessKey="a,b,c,d" 1`] = `
<tbody
  accessKey="a,b,c,d"
/>
`;

exports[`tbody renders with autoCapitalize="characters" 1`] = `
<tbody
  autoCapitalize="characters"
/>
`;

exports[`tbody renders with autoCapitalize="none" 1`] = `
<tbody
  autoCapitalize="none"
/>
`;

exports[`tbody renders with autoCapitalize="off" 1`] = `
<tbody
  autoCapitalize="off"
/>
`;

exports[`tbody renders with autoCapitalize="on" 1`] = `
<tbody
  autoCapitalize="on"
/>
`;

exports[`tbody renders with autoCapitalize="sentences" 1`] = `
<tbody
  autoCapitalize="sentences"
/>
`;

exports[`tbody renders with autoCapitalize="words" 1`] = `
<tbody
  autoCapitalize="words"
/>
`;

exports[`tbody renders with children 1`] = `
<tbody>
  TEST
</tbody>
`;

exports[`tbody renders with className="foo bar baz" 1`] = `
<tbody
  className="foo bar baz"
/>
`;

exports[`tbody renders with contentEditable="false" 1`] = `
<tbody
  contentEditable="false"
/>
`;

exports[`tbody renders with contentEditable="true" 1`] = `
<tbody
  contentEditable="true"
/>
`;

exports[`tbody renders with dir="auto" 1`] = `
<tbody
  dir="auto"
/>
`;

exports[`tbody renders with dir="ltr" 1`] = `
<tbody
  dir="ltr"
/>
`;

exports[`tbody renders with dir="rtl" 1`] = `
<tbody
  dir="rtl"
/>
`;

exports[`tbody renders with draggable="false" 1`] = `
<tbody
  draggable="false"
/>
`;

exports[`tbody renders with draggable="true" 1`] = `
<tbody
  draggable="true"
/>
`;

exports[`tbody renders with empty attributes 1`] = `<tbody />`;

exports[`tbody renders with hidden="false" 1`] = `
<tbody
  hidden={false}
/>
`;

exports[`tbody renders with hidden="true" 1`] = `
<tbody
  hidden={true}
/>
`;

exports[`tbody renders with id="TEST" 1`] = `
<tbody
  id="TEST"
/>
`;

exports[`tbody renders with inputMode="decimal" 1`] = `
<tbody
  inputMode="decimal"
/>
`;

exports[`tbody renders with inputMode="email" 1`] = `
<tbody
  inputMode="email"
/>
`;

exports[`tbody renders with inputMode="none" 1`] = `
<tbody
  inputMode="none"
/>
`;

exports[`tbody renders with inputMode="numeric" 1`] = `
<tbody
  inputMode="numeric"
/>
`;

exports[`tbody renders with inputMode="search" 1`] = `
<tbody
  inputMode="search"
/>
`;

exports[`tbody renders with inputMode="tel" 1`] = `
<tbody
  inputMode="tel"
/>
`;

exports[`tbody renders with inputMode="text" 1`] = `
<tbody
  inputMode="text"
/>
`;

exports[`tbody renders with inputMode="url" 1`] = `
<tbody
  inputMode="url"
/>
`;

exports[`tbody renders with is="foo" 1`] = `
<tbody
  is="foo"
/>
`;

exports[`tbody renders with itemID="https://google.com/" 1`] = `
<tbody
  itemID="https://google.com/"
/>
`;

exports[`tbody renders with itemProp="https://google.com/" 1`] = `
<tbody
  itemProp="https://google.com/"
/>
`;

exports[`tbody renders with itemRef="foo bar" 1`] = `
<tbody
  itemRef="foo bar"
/>
`;

exports[`tbody renders with itemScope="false" 1`] = `
<tbody
  itemScope={false}
/>
`;

exports[`tbody renders with itemScope="true" 1`] = `
<tbody
  itemScope={true}
/>
`;

exports[`tbody renders with itemType="https://google.com/" 1`] = `
<tbody
  itemType="https://google.com/"
/>
`;

exports[`tbody renders with lang="" 1`] = `
<tbody
  lang=""
/>
`;

exports[`tbody renders with lang="es-419" 1`] = `
<tbody
  lang="es-419"
/>
`;

exports[`tbody renders with slot="TEST" 1`] = `
<tbody
  slot="TEST"
/>
`;

exports[`tbody renders with spellCheck="false" 1`] = `
<tbody
  spellCheck="false"
/>
`;

exports[`tbody renders with spellCheck="true" 1`] = `
<tbody
  spellCheck="true"
/>
`;

exports[`tbody renders with style="[object Object]" 1`] = `
<tbody
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`tbody renders with tabIndex="1" 1`] = `
<tbody
  tabIndex={1}
/>
`;

exports[`tbody renders with title="TEST" 1`] = `
<tbody
  title="TEST"
/>
`;

exports[`tbody renders with translate="no" 1`] = `
<tbody
  translate="no"
/>
`;

exports[`tbody renders with translate="yes" 1`] = `
<tbody
  translate="yes"
/>
`;

exports[`tbody renders without attributes 1`] = `<tbody />`;
